Casks/w/wordpresscom.rb

```diff
@@ -1,23 +1,43 @@
 cask "wordpresscom" do
+  arch arm:   "arm64",
+       intel: "x64"
+
   version "8.0.3"
-  sha256 "cc5f9daccc774eb9a3fbd3533e26357ff3dd226aa95355028fdfb1d745b6619e"
+  sha256 arm:   "fa918e3df870deafe299b11a16b3753f6fb8a1af36b0c616620c1c1b98e20ec8",
+         intel: "98f54a7559b57146dc3689c83609d82f0c57fa0f2576d5a84b38b18ea1e2bec7"
 
-  url "https://public-api.wordpress.com/rest/v1.1/desktop/osx/download?type=app&ref=update&version=#{version}"
+  url "https://github.com/Automattic/wp-desktop/releases/download/v#{version}/wordpress.com-macOS-dmg-#{version}-#{arch}.dmg", verified: "github.com/Automattic/wp-desktop/"
   name "WordPress.com"
   desc "WordPress client"
   homepage "https://apps.wordpress.com/desktop/"
 
   livecheck do
-    url "https://public-api.wordpress.com/rest/v1.1/desktop/osx/download?type=dmg"
-    strategy :header_match
+    url "https://github.com/Automattic/wp-desktop"
+    strategy :github_releases do |json, regex|
+      json.map do |release|
+        next if release["draft"] || release["prerelease"]
+
+        match = release["tag_name"]&.match(regex)
+        next if match.blank?
+
+        match[1]
+      end
+    end
```

Comment:
This requires a regex to be passed. I think we should be able to try using the below instead, unless upstream have releases without MacOS related binaries.

```suggestion
    url :url
    strategy :github_latest
```

Reply:
Good catch with `url :url`. Thanks!

Does `:github_latest` skip betas/pre-releases? This is unclear to me from the [livecheck docs](https://docs.brew.sh/Brew-Livecheck#githublatest-strategy-block).

Upstream does publish betas, as well as packages for Linux and Windows, at that [GitHub Releases url](https://github.com/Automattic/wp-desktop/releases).

I was under the impression that

1. `:github_latest` does not skip betas, and the only way to do that is as in my initial commit, and
3. that, absent a specified regex, livecheck will resort to a fallback regex.

Reply:
`:github_latest` will use the release that is marked as "latest" in the repository.
If upstream mark pre-releases/betas as "latest" then it will not work.

Reply:
Is there any way to find out if they have done that, historically, after-the-fact? Otherwise, it seems this PR could languish until upstream decides to release another beta.